app/main/views/communications.py

```diff
@@ -1,37 +1,69 @@
+from pathlib import PurePath
+
 from dmutils import s3  # this style of import so we only have to mock once
-from dmutils.documents import file_is_pdf, file_is_csv, file_is_open_document_format
+from dmutils.documents import file_is_pdf, file_is_csv, file_is_open_document_format, get_signed_url
 from dmutils.flask import timed_render_template as render_template
-from flask import redirect, url_for, current_app, request, flash
+from flask import redirect, url_for, current_app, request, flash, abort
 
 from .. import main
 from ..auth import role_required
 from ... import data_api_client
 
 
+def _get_comm_type_root(framework_slug, comm_type):
+    # PurePath has the advantage that it doesn't handle ".." elements specially, and so we're safe from a whole
+    # category of tricks
+    return PurePath(framework_slug) / "communications" / "updates" / f"{comm_type}s"
+
+
+_comm_types = ("communication", "clarification",)
```

Comment:
This might be overkill, but what about making this an enum type?

Reply:
Because I already feel like I've overcooked this story.